packages/circuits/crates/libs/polynomial/src/lib.nr

```diff
@@ -98,38 +98,87 @@ impl<let N: u32> Polynomial<N> {
     }
 }
 
-/// Flattens an array of polynomials into a 1D array of Field elements.
-///
-/// This helper function serializes polynomial coefficients maintaining a deterministic
-/// ordering of coefficients to ensure reproducible challenge values.
-///
-/// # Generic Parameters
-/// * `A` - The degree of each polynomial in the array
-/// * `L` - The number of polynomials in the array
-/// * `SIZE` - The total size of the target array
-///
-/// # Arguments
-/// * `inputs` - The target array to store flattened coefficients
-/// * `poly` - Array of polynomials to flatten
-/// * `offset` - Starting position in the inputs array
-///
-/// # Returns
-/// A tuple containing:
-/// * The updated inputs array with flattened coefficients
-/// * The new offset position after flattening
-pub fn flatten<let A: u32, let L: u32, let SIZE: u32>(
-    mut inputs: [Field; SIZE],
+/// Flatten `L` polynomials (each `Polynomial<A>`) into a linear Vec<Field> of packed carriers.
+/// - Uses the same hex-aligned packing as `packer::<A, BIT>`:
+///   `nibble_bits = ceil_to_nibble(BIT)`, `limb_bits = nibble_bits + 4`,
+///   `group = 254 / limb_bits` carriers per chunk.
+/// - Preserves deterministic order: iterate polynomials in `j`, then append all their carriers.
+pub fn flatten<let A: u32, let L: u32, let BIT: u32>(
+    mut inputs: Vec<Field>,
     poly: [Polynomial<A>; L],
-    mut offset: u32,
-) -> ([Field; SIZE], u32) {
+) -> Vec<Field> {
+    // Compute packing layout (must mirror `packer`).
+    let nibble_bits = ((BIT + 3) / 4) * 4; // ceil to multiple of 4
+    let limb_bits = nibble_bits + 4; // +1 nibble offset
```

Comment:
like, this might be a bit cryptic to read later on if need to make further changes

Reply:
like someone might ask why doing `() / 4) * 4`